internal/collection/workflow.go

```diff
@@ -30,18 +29,31 @@ type ProcessingWorkflowRequest struct {
 	// an existing collection in retries.
 	CollectionID uint
 
-	// Captured by the watcher, the event contains information about the
-	// incoming dataset.
-	Event *watcher.BlobEvent
+	WatcherName string
+
+	PipelineName string
+
+	RetentionPeriod *time.Duration
+
+	StripTopLevelDir bool
+
+	Key string
+
+	BatchDir string
```

Comment:
Some questions and thoughts:
* Is `Key` is for objects and `BatchDir` is for files/dirs locally available? Both files and dirs are supported in batching mode right?
* Instead of `BatchDir`, should we call it `LocalPath`?
* Can you remind me why we decided to take this route instead of using a single URL attribute with `s3` and `file` schemes?
* I see the docs in `TransferInfo` but I suggest to do the same here, I think `ProcessingWorkflowRequest` matters because it's the user-facing type.

Reply:
> Is `Key `is for objects and `BatchDir` is for files/dirs locally available? Both files and dirs are supported in batching mode right?

`Key` can be a filename or a directory name (if it's part of a `BatchDir`). At the moment batches only support directories because if the batch is in the transfer directory its contents are not supposed to be moved. I plan to file an issue describing this problem.

> Can you remind me why we decided to take this route instead of using a single URL attribute with `s3` and `file` schemes?

I think this would require modifying watchers? Which is something I didn't want to do in this iteration. When we discussed this idea I started a [`dev/buckets`](https://github.com/artefactual-labs/enduro/tree/dev/buckets) branch with the goal of treating batches as buckets.

> I see the docs in `TransferInfo` but I suggest to do the same here, I think `ProcessingWorkflowRequest` matters because it's the user-facing type.

What would you recommend to not duplicate documentation of request parameters that also become part of `TransferInfo`?